contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt

```diff
@@ -159,6 +159,80 @@ data class PlacedHexTiles @ConstructorForDeserialization constructor(val value:
+        if (opposite == null || side.tileIndex.value <= opposite.tileIndex.value) side
+        else opposite
+    }
+
+    fun getSideHash(side: AbsoluteSide) = getSmallerEquivalent(side).let {
+        it.tileIndex.value * HexTile.SIDE_COUNT + it.sideIndex.value
+    }
+
+    /**
+     * When opposite sides, then the one with the lower tile index is smaller.
+     * When different sides, then, with the opposites, the one with the lower tile index is smaller,
+     * and if equal, the one with the smaller side index is smaller.
+     */
+    fun getSideComparator() = Comparator<AbsoluteSide> { left, right ->
+        if (left == null && right == null) return@Comparator 0
+        if (left == null) return@Comparator -1
+        if (right == null) return@Comparator 1
+        val leftEq = getSmallerEquivalent(left)
+        val rightEq = getSmallerEquivalent(right)
+        if (left == right) 0
+        else if (leftEq == rightEq) left.tileIndex.value - right.tileIndex.value
+        else (leftEq.tileIndex.value - rightEq.tileIndex.value).let {
+            if (it != 0) it
+            else leftEq.sideIndex.value - rightEq.sideIndex.value
+        }
+    }
+
+    fun getOverlappedCorners(absoluteCorner: AbsoluteCorner) = get(absoluteCorner.tileIndex)
+            .sides
+            .getNeighborsOn(absoluteCorner.cornerIndex)
+            .zip(absoluteCorner.cornerIndex.getOverlappedCorners())
+            .map { AbsoluteCorner.create(it.first, it.second) }
+
+    /**
+     * When 1, 2 or 3 AbsoluteCorners are neighbors, the smaller equivalent is the one with the lower tileIndex.
+     */
+    fun getSmallestEquivalent(corner: AbsoluteCorner) = getOverlappedCorners(corner)
+            .fold(corner) { choice, next ->
+                if (next == null || choice.tileIndex.value <= next.tileIndex.value) choice
+                else next
+            }
+
+    fun getCornerHash(corner: AbsoluteCorner) = getSmallestEquivalent(corner).let {
+        it.tileIndex.value * HexTile.SIDE_COUNT + it.cornerIndex.value
+    }
+
+    /**
+     * When overlapping corner, then they are ordered by their tile index.
+     * When different corners, then, they are ordered by the tile order of their smallest overlapping ones,
+     * and if equal, the one with the smaller side index is smaller.
+     */
+    fun getCornerComparator() = Comparator<AbsoluteCorner> { left, right ->
+        if (left == null && right == null) return@Comparator 0
+        if (left == null) return@Comparator -1
+        if (right == null) return@Comparator 1
+        val leftEq = getSmallestEquivalent(left)
+        val rightEq = getSmallestEquivalent(right)
+        if (left == right) 0
+        else if (leftEq == rightEq) left.tileIndex.value - right.tileIndex.value
+        else (leftEq.tileIndex.value - rightEq.tileIndex.value).let {
+            if (it != 0) it
+            else leftEq.cornerIndex.value - rightEq.cornerIndex.value
+        }
+    }
+
```

Comment:
@abellagonzalo Have a look at the functions above. They should help you traverse the graph in search of the longest road.

Reply:
Look good. I like the idea of the corner and side comparators.

Reply:
@abellagonzalo If you find yourself manipulating sides and corners other than traversing and keeping tabs on them, let me know, so I can help devise yet another function.
Also, make a small pull request with a minor change so that you make sure that process is working for you.

Reply:
That was my first commit https://github.com/rogersanick/SettlersOfCordan/commit/404727b019bba63a6634698fea3e9a5ee0d8daf4